### ISSUE TITLE
Fails to compile with gcc 11.2.0

### ISSUE THREAD

**Issue by mdruby** [JohannesBuchner/cuba#8]

**Comment by mdruby** [comment]
Here's the output when running ./makesharedlib.sh.  Looks like it may be related to this:
https://github.com/Shopify/semian/issues/261#issuecomment-781685569

gcc -shared -Wall Vegas.o Vegas_.o llVegas.o llVegas_.o Suave.o Suave_.o llSuave.o llSuave_.o Divonne.o Divonne_.o llDivonne.o llDivonne_.o Cuhre.o Cuhre_.o llCuhre.o llCuhre_.o Fork.o Fork_.o Global.o Global_.o Data.o -lm -o libcuba.so
/usr/bin/ld: Vegas_.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: llVegas.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: llVegas_.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: Suave.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: Suave_.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: llSuave.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: llSuave_.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: Divonne.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: Divonne_.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: llDivonne.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: llDivonne_.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: Cuhre.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: Cuhre_.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: llCuhre.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: llCuhre_.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: Fork.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: Fork_.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: Global.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
/usr/bin/ld: Global_.o:(.bss+0x0): multiple definition of `cubafun_'; Vegas.o:(.bss+0x0): first defined here
collect2: error: ld returned 1 exit status

**Comment by JohannesBuchner** [comment]
Why did you close this issue, did you find a solution?

I don't think I can edit the code, because I did not write it. Maybe we should upgrade cuba to the latest version, and see if we can get the python interface to work there.

**Comment by mdruby** [comment]
Hi, I realized it was a repeat of a previous issue. I followed your advice of adding -fcommon to line 5 of the makesharedlib script. After running make clean, it worked fine.

Sent from my iPad

On Apr 1, 2022, at 1:50 AM, Johannes Buchner ***@***.***> wrote:

﻿

Why did you close this issue, did you find a solution?

I don't think I can edit the code, because I did not write it. Maybe we should upgrade cuba to the latest version, and see if we can get the python interface to work there.
